Add OptionLogAllOutput

JesseTG opened the pull request.

JesseTG committed:
Add OptionLogAllOutput

sttz commented:
With the Addressables option being in the Build category, "Log All Output" implies to me that it's related to the build process. Maybe calling it "Enable Full Log In Build" (or "Enable Log All in Build" to be more in line with the define) would be more clear?

JesseTG commented:
Ah, yeah, that makes sense. I'll go ahead and make that change today.

JesseTG committed:
Rename OptionLogAllOutput to OptionEnableRuntimeLogging

- Clarifies the runtime effects of this option in line with the maintainer's suggestion
